src/main/scala/rps/Game.scala

```diff
@@ -1,30 +1,40 @@
 package rps
 
+import RPSMoves._
+
 object Game {
   def play(): Unit = {
 
     /* Ask for user input */
-    val userMove = readLine(s"""
+    val userMoveString = readLine(s"""
     |What's your move?"
-    |0: ${returnMoveIcon("0")}
-    |1: ${returnMoveIcon("1")}
-    |2: ${returnMoveIcon("2")}
+    |0: ${ROCK.toString}
+    |1: ${PAPER.toString}
+    |2: ${SCISSORS.toString}
     |
     |> """.stripMargin)
 
-    val computerMove = generateCPUMove();
+    val userMove = RPSMoves.factoryMove(userMoveString)
+
+    if (userMove == None) {
```

Comment:
When you have an `Option` you generally don't use `if/else` to work with the two possible cases. In this case, I would suggest to use a `match` (which will save you the `.get` below)

Reply:
Very useful! Thanks a lot! I was getting horrified at all those .get!